### PR TITLE
ACAS-704 test project restrictions

### PR DESCRIPTION
**Pull request by brianbolt**
## Description
 - Adding tests for experiment project restrictions

## Related Issue
ACAS-712

## How Has This Been Tested?
Ran acasclient tests

### CODE REVIEW

**Review by dalejerikson — APPROVED** [review]
LGTM!

**Comment by dalejerikson** [comment]
I see that in GitHub Actions that there's one error but I'm not sure if there's still problems with ACAS builds in Github or not...

`File "/home/runner/work/acasclient/acasclient/acasclient/tests/test_acasclient.py", line 3346, in test_004_delete_lot
    self.assertTrue(self.check_lot_exists_in_experiment(self, restricted_lot_corp_name, response['results']['experimentCode']))
TypeError: check_lot_exists_in_experiment() takes 3 positional arguments but 4 were given`

File: tests/test_acasclient.py [hunk 1]
```diff
@@ -3343,7 +3343,7 @@ def check_lot_exists_in_experiment(self, lot_corp_name, experiment_code_name):
         file_to_upload = get_basic_experiment_load_file(self.tempdir, project.names[PROJECT_NAME], restricted_lot_corp_name, file_name='4 parameter D-R.csv')
         response = self.client.\
             experiment_loader(file_to_upload, "bob", False)
-        self.assertTrue(check_lot_exists_in_experiment(self, restricted_lot_corp_name, response['results']['experimentCode']))
+        self.assertTrue(self.check_lot_exists_in_experiment(self, restricted_lot_corp_name, response['results']['experimentCode']))
```
Comment (iamanandkr) [inline tests/test_acasclient.py:3346]: `self` shouldn't be passed now. It's the same for line 3362.

**Review by iamanandkr — DISMISSED** [review]
Thank you Brian! 
I only have comment about the failing test, other than that LGTM!

File: tests/test_acasclient.py [hunk 2]
```diff
@@ -4600,4 +4600,51 @@ def test_018_image_file(self):
...
+        # (Allowed) Restricted experiment proj1 with lot unrestricted
+        # Basic cmpdreg load creates a lot in the Global unrestricted project
+        # So we expect that it can be uploaded to a restricted project
+        # Get a file to upload to restricted_project_1 but has a lot from the Global unrestricted project
+        file_to_upload = get_basic_experiment_load_file(self.tempdir, restricted_project_1.names[PROJECT_NAME], corp_name=None, file_name='uniform-commas-with-quoted-text.csv')
+        # Upload the file and expect it to succeed
+        response = self.experiment_load_test(file_to_upload, True, expect_failure=False)
+
+        # (Allowed) Unrestricted experiment proj1 with lot restricted proj2
+        # Create an upload file which uploads to restricted_project_1 but has a lot from restricted_project_2
```
Comment (bffrost) [inline tests/test_acasclient.py:4647]: From the comment this looks like a good test to run. However I'm confused about your implementation. On the line below this, you seem to be loading a lot in restricted project 1 into experiment restricted project 1.
I don't see you create an unrestricted experiment project or using Global as the experiment project.
Also the lot seems to be from project 1 but your comment here says it's from project 2.

**Review by bffrost — CHANGES_REQUESTED** [review]
Great tests to have. Last part of test 19 doesn't seem to line up with the code comment. Also see Anand's comment to fix the breaking test

File: tests/test_acasclient.py (hunk 2 above)
Comment (iamanandkr) [reply tests/test_acasclient.py]: I have addressed the comment. Let me know if now the test seems correct to you!

**Review by bffrost — APPROVED** [review]
LGTM

File: tests/test_acasclient.py (hunk 1 above)
Comment (brianbolt) [reply tests/test_acasclient.py]: Ah! Thanks.

File: tests/test_acasclient.py (hunk 2 above)
Comment (brianbolt) [reply tests/test_acasclient.py]: Thanks. LGTM.  I changed the test and forgot to fix the comment.

**Pull request merged**